### ISSUE TITLE
rx-scavenger: Doesn't work without OnDestroy

### ISSUE THREAD

**Issue by CanKattwinkel** [wishtack/wishtack-steroids#146]
I was curious since according to the readme this should work without implementing the OnDestroy interface. But I tried it and it doesn't work. 

The lib fails because Angular doesn't call the OnDestroy lifecycle method when the method is added afterwards. In order to fix this you are probably required to change `component?: ComponentWithOptionalOnDestroy` to `component?: OnDestroy`

See this discussion for more details:
https://github.com/w11k/ng2-rx-componentdestroyed/pull/1

**Comment by yjaaidi** [comment]
OMG! I didn't check that!
Thank you for the suggestion! I just applied it but I find it annoying to implement the `ngOnDestroy` all the time :/.

The other solution I had in mind, is to use `Scavenger` as a service that should be provided by the component itself and the service would unsubscribe when the service's `ngOnDestroy` method is called.
```ts
@Component({
    providers: [Scavenger]
})
export class CounterComponent {
    constructor(privarte _scavenger: Scavenger) {}
}
```
Would that be nice?

I am worried about the error-prone aspect of this as we should always declare the Scavenger in all components' providers :/

**Comment by romanroe** [comment]
@yjaaidi Unfortunately, I dont see much value of 

```
...
providers: [Scavenger]
...
constructor(privarte _scavenger: Scavenger)
```

compared to 

```
ngOnDestroy() {}
```

This solution https://github.com/w11k/ng2-rx-componentdestroyed (note: I am the author) at least ensures that the component implements the method stub, you can't do it wrong.

**Comment by yjaaidi** [comment]
Hi @romanroe!

Yup! You are right!

The only other workaround I found is to get an instance of `ChangeDetectorRef` and downcast it to `ViewRef` (which I consider too dirty to be acceptable) then use `View.onDestroy` method to add our callback. I tried it. It works but I find this solution too dirty to be acceptable.

I wonder how things are going to change with Ivy renderer.

Meanwhile, I can't find any cleaner solution :/
